dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -11,129 +11,292 @@
     {%- endfor -%})))
 {%- endmacro %}
 
+
 {%
-  Recreate the snapshot table from the new_snapshot_table
+    If hive table then Recreate the snapshot table from the new_snapshot_table
+    If iceberg table then Update the standard snapshot merge to include the DBT_INTERNAL_SOURCE prefix in the src_cols_csv
 %}
 
-{% macro athena__snapshot_merge_sql(target, source) -%}
-    {%- set target_relation = adapter.get_relation(database=target.database, schema=target.schema, identifier=target.identifier) -%}
-    {%- if target_relation is not none -%}
-      {% do adapter.drop_relation(target_relation) %}
-    {%- endif -%}
+{% macro athena__snapshot_merge_sql(target, source, insert_cols, table_type) -%}
+    {% if table_type=='iceberg' %}
+        {%- set insert_cols_csv = insert_cols | join(', ') -%}
+        {%- set src_columns = [] -%}
+        {%- for col in insert_cols -%}
+          {%- do src_columns.append('DBT_INTERNAL_SOURCE.' + col) -%}
+        {%- endfor -%}
+        {%- set src_cols_csv = src_columns | join(', ') -%}
+
+        merge into {{ target }} as DBT_INTERNAL_DEST
+        using {{ source }} as DBT_INTERNAL_SOURCE
+        on DBT_INTERNAL_SOURCE.dbt_scd_id = DBT_INTERNAL_DEST.dbt_scd_id
```

Comment:
as above, and as per repo standards, could you please use lowercase everywhere? e.g. column names or table aliases.